### PR TITLE
Update sample app to use latest insights client package

### PR DESCRIPTION
**Pull request by bingjiez**

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -14,40 +14,41 @@
...
+    "@bentley/ui-abstract": "^2.19.51",
+    "@bentley/webgl-compatibility": "^2.19.51",
     "electron": "^11.1.0"
   },
   "devDependencies": {
-    "@bentley/build-tools": "^2.17.2",
-    "@bentley/eslint-plugin": "^2.17.2",
-    "@types/chai": "^4.2.21",
-    "@types/mocha": "^8.2.3",
+    "@bentley/build-tools": "2.19.51",
```
Comment (MindaugasButkus) [inline package.json:41]: Upgrade to latest iTwinJs version (3.6.1)

**Comment by Guillar1** [comment]
The repo PR policies checks don't seem to be setup correctly.

File: src/Main.ts [hunk 2]
```diff
@@ -7,15 +7,11 @@ import { assert, BentleyError, BentleyStatus } from "@bentley/bentleyjs-core";
...
-import { GroupProperty, GroupPropertiesClient, GroupPropertyUpdateParams } from "./GroupPropertiesClient";
-import { Mapping, MappingsClient } from "./MappingsClient";
-import { Report, ReportsClient } from "./ReportsClient";
-import { ExtractionClient, Run, Status, ExtractionRunState } from './ExtractionClient';
-import { ReportMapping, ReportMappingsClient } from "./ReportMappingsClient";
+import { AccessToken } from "@bentley/itwin-client";
+import { CalculatedProperty, CalculatedPropertyCreate, CustomCalculation, CustomCalculationCreate, ExtractionClient, ExtractionStatus, ExtractorState, Group, GroupProperty, GroupPropertyCreate, IExtractionClient, IMappingsClient, IReportsClient, Mapping, MappingsClient, Report, ReportMapping, ReportsClient } from "@itwin/insights-client";
 
 const config = require('./config.json');
+const BASE_URL = "https://api.bentley.com/insights/reporting";
```
Comment (Guillar1) [inline src/Main.ts:14]: Is this `BASE_URL` needed?

File: package.json [hunk 3]
```diff
@@ -14,40 +14,24 @@
...
-    "@bentley/product-settings-client": "^2.17.2",
-    "@bentley/rbac-client": "^2.17.2",
-    "@bentley/telemetry-client": "^2.17.2",
-    "@bentley/ui-abstract": "^2.17.2",
-    "@bentley/webgl-compatibility": "^2.17.2",
-    "electron": "^11.1.0"
+    "@itwin/insights-client": "0.4.0",
+    "@itwin/core-bentley": "^3.6.1",
+    "@itwin/core-backend": "^3.6.1",
+    "@itwin/service-authorization": "^0.7.0",
```
Comment (MindaugasButkus) [inline package.json:20]: @itwin/node-cli-authorization might be a better fit for this. With service-authorization you might need to invite the "service account" into your iTwin for it to be able to do anything.

Comment (MindaugasButkus) [reply package.json:20]: With "node-cli-authorization" package users would need to create a "Desktop/Mobile" application which does not use client secret.

File: package.json [hunk 4]
```diff
@@ -14,40 +14,24 @@
...
-    "@bentley/rbac-client": "^2.17.2",
-    "@bentley/telemetry-client": "^2.17.2",
-    "@bentley/ui-abstract": "^2.17.2",
-    "@bentley/webgl-compatibility": "^2.17.2",
-    "electron": "^11.1.0"
+    "@itwin/insights-client": "0.4.0",
+    "@itwin/core-bentley": "^3.6.1",
+    "@itwin/core-backend": "^3.6.1",
+    "@itwin/node-cli-authorization": "^0.10.0",
+    "electron": "^23.1.2"
```
Comment (MindaugasButkus) [inline package.json:21]: `electron` should no longer be needed.

File: package.json [hunk 5]
```diff
@@ -14,40 +14,24 @@
...
+    "@bentley/eslint-plugin": "2.19.51",
+    "@types/chai": "^4.1.4",
+    "@types/mocha": "^8.2.2",
     "@types/sinon": "^10.0.2",
-    "chai": "^4.3.4",
-    "eslint": "^6.8.0",
-    "mocha": "^9.0.2",
+    "chai": "^4.1.2",
+    "eslint": "^7.11.0",
+    "mocha": "^8.3.2",
```
Comment (MindaugasButkus) [inline package.json:31]: There are no tests in this app, so `mocha`, `chai`, `@types/chai`, `@types/mocha`, `@types/sinon`, `sinon` are not needed.

Comment (MindaugasButkus) [reply package.json]: Remove `test` and and `copy:test-assets` scrips as well.

Comment (MindaugasButkus) [reply package.json]: Remove `mocha` configuration from package.json.

File: package.json (hunk 1 above)
Comment (MindaugasButkus) [reply package.json]: This is done, but I do not have a "resolve" button for any of the comments I added :/

Comment (bingjiez) [reply package.json]: I'll do it 🚀

File: src/Main.ts (hunk 2 above)
Comment (bingjiez) [reply src/Main.ts]: removed

File: package.json [hunk 6]
```diff
@@ -9,45 +9,20 @@
...
-    "mocha": "^9.0.2",
+    "@itwin/build-tools": "3.6.1",
+    "@itwin/eslint-plugin": "3.6.1",
+    "eslint": "^7.11.0",
     "rimraf": "^3.0.2",
-    "sinon": "^11.1.1",
-    "typescript": "^3.7.7"
+    "typescript": "~4.3.0"
   },
   "eslintConfig": {
```
Comment (MindaugasButkus) [inline package.json:27]: `eslintConfig` looks outdated. Update according to https://www.npmjs.com/package/@itwin/eslint-plugin readme.

Comment (bingjiez) [reply package.json:28]: Done.

File: src/Main.ts (hunk 2 above)
Comment (Guillar1) [reply src/Main.ts]: I can't resolve this for some reason.

Comment (bingjiez) [reply src/Main.ts]: should be able now

File: package.json (hunk 6 above)
Comment (bingjiez) [reply package.json:28]: I guess we can close this PR?

**Pull request merged**